gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -56,9 +58,10 @@
   }
 
   @Override
-  public Map<String, String> getPathParams(ApiMessage apiMessage, Set<String> paramNames) {
+  public Map<String, String> getPathParams(T apiMessage, Set<String> paramNames) {
+    Struct resourceName = (Struct) apiMessage.getResourceName();
```

Comment:
This doesn't seem like safe cast to make - it seems like the resource name could be set to something that's not a Struct.

Reply:
Ok, removed that cast!